keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/StepsizeTest.java

```diff
@@ -0,0 +1,103 @@
+
+    @Test
+    public void canFindSmallStartingStepsizeForSmallSpace() {
+        DoubleVertex vertex = new GaussianVertex(0, 0.05);
+        List<DoubleVertex> vertices = Arrays.asList(vertex);
+        BayesianNetwork bayesianNetwork = new BayesianNetwork(vertex.getConnectedGraph());
+
+        VertexId vertexId = vertex.getId();
+
+        LogProbGradientCalculator logProbGradientCalculator = new LogProbGradientCalculator(bayesianNetwork.getLatentOrObservedVertices(), vertices);
+        vertex.setValue(DoubleTensor.scalar(1.));
+        Map<VertexId, DoubleTensor> position = Collections.singletonMap(vertexId, vertex.getValue());
+        Map<VertexId, DoubleTensor> gradient = logProbGradientCalculator.getJointLogProbGradientWrtLatents();
+
+        double startingStepsize = Stepsize.findStartingStepSize(
+            position,
+            gradient,
+            Arrays.asList(vertex),
+            bayesianNetwork.getLatentVertices(),
+            logProbGradientCalculator,
+            ProbabilityCalculator.calculateLogProbFor(vertices),
+            random
+        );
+
+        double startingEpsilon = 1.0;
+        Assert.assertTrue(startingStepsize < startingEpsilon);
+    }
+
+    @Test
+    public void canFindLargeStartingStepsizeForLargeSpace() {
+        DoubleVertex vertex = new GaussianVertex(0, 500.);
+        List<DoubleVertex> vertices = Arrays.asList(vertex);
+        BayesianNetwork bayesianNetwork = new BayesianNetwork(vertex.getConnectedGraph());
+
+        VertexId vertexId = vertex.getId();
+
+        LogProbGradientCalculator logProbGradientCalculator = new LogProbGradientCalculator(bayesianNetwork.getLatentOrObservedVertices(), vertices);
+        Map<VertexId, DoubleTensor> position = Collections.singletonMap(vertexId, vertex.sample(random));
+        Map<VertexId, DoubleTensor> gradient = logProbGradientCalculator.getJointLogProbGradientWrtLatents();
+
+        double startingStepsize = Stepsize.findStartingStepSize(
+            position,
+            gradient,
+            Arrays.asList(vertex),
+            bayesianNetwork.getLatentVertices(),
+            logProbGradientCalculator,
+            ProbabilityCalculator.calculateLogProbFor(vertices),
+            random
+        );
+
+        Assert.assertTrue(startingStepsize > 64);
+    }
+
+    @Test
+    public void canReduceStepsizeFromLargeInitialToSmallToExploreSmallSpace() {
+        double startingStepsize = 10.;
+
+        Stepsize tune = new Stepsize(
+            startingStepsize,
+            0.65,
+            50
+        );
+
+        TreeBuilder mockedLessLikelyTree = mock(TreeBuilder.class);
+        when(mockedLessLikelyTree.getDeltaLikelihoodOfLeapfrog()).thenAnswer(i -> -50.);
+        when(mockedLessLikelyTree.getTreeSize()).thenAnswer(i -> 8.);
+        double adaptedStepSizeLessLikely = tune.adaptStepSize(mockedLessLikelyTree, 1);
+
+        Assert.assertTrue(adaptedStepSizeLessLikely < startingStepsize);
```

Comment:
I'm not sure how valuable this is, but you can verify the methods that were called on your mocks:

        verify(mockedLessLikelyTree).getDeltaLikelihoodOfLeapfrog();
        verify(mockedLessLikelyTree).getTreeSize();
        verifyNoMoreInteractions(mockedLessLikelyTree);